### PR TITLE
doc: update EVP_PKEY_size() documentation to indicate that its is unreliable

### LINKED ISSUE #15552: EVP_PKEY_size() returning too small value for a signature
While testing libssh, it was found that the following call [1] to EVP_DigestSign() overwrites raw_sig_len.
Using gdb, I checked the value of raw_sig_len before passing it to EVP_DigestSign().
It's value was 32 (as assigned in [2])
After passing it to EVP_DigestSign(),
the value is 64.
[1]https://gitlab.com/libssh/libssh-mirror/-/blob/master/src/pki_crypto.c#L2236
[2] https://gitlab.com/libssh/libssh-mirror/-/blob/master/src/pki_crypto.c#L2212

Because of this memory corruption, there is a crash at [3].

[3] https://gitlab.com/libssh/libssh-mirror/-/blob/master/src/pki_crypto.c#L2296

(as tested with latest master 3.0.0-beta1-dev)

If I change the value of raw_sig_data [4] to
raw_sig_data = (unsigned char *)malloc(4*raw_sig_len);
then the crash is no longer seen.

[4] https://gitlab.com/libssh/libssh-mirror/-/blob/master/src/pki_crypto.c#L2213

The same test has been passing with OpenSSL version 1.1.1.

Kindly investigate this further. Thank you.

### PR DESCRIPTION
**Pull request by paulidale**
Fixes #15552

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by beldmit — APPROVED** [review]
LGTM

File: doc/man3/EVP_PKEY_size.pod [hunk 1]
```diff
@@ -25,10 +25,10 @@ L<EVP_PKEY_encrypt(3)>, L<EVP_PKEY_decrypt(3)>, L<EVP_PKEY_derive(3)>.
...
 that's being performed says otherwise, the size returned by
 EVP_PKEY_size() is only preliminary and not exact, so the final
-contents of the target buffer may be smaller.  It is therefore crucial
-to take note of the size given back by the function that performs the
-operation, such as L<EVP_PKEY_sign(3)> (the I<siglen> argument will
-receive that length), to avoid bugs.
+contents of the target buffer may be smaller or larger than indicated.
+It is therefore crucial to take note of the size given back by the
+function that performs the operation, such as L<EVP_PKEY_sign(3)>
+(the I<siglen> argument will receive that length), to avoid bugs.
```
Comment (t8m) [inline doc/man3/EVP_PKEY_size.pod:31]: I am not sure this is right change. The first paragraph is completely invalidated by this change. Which algorithms produce signatures bigger than EVP_PKEY_size? And shouldn't the fix be to actually change the EVP_PKEY_size to return something to conform to its documentation?

Comment (t-j-h) [reply doc/man3/EVP_PKEY_size.pod:31]: No! They cannot be larger. Must be equal to or smaller than. Not larger.

**Comment by mattcaswell** [comment]
This is the wrong answer to this problem. Fix `EVP_PKEY_size()` so that it behave in the same way as 1.1.1.

**Pull request closed without merging**